### DISCUSSION TITLE
Getting nonsense hexabyte from PMD after using client.write_gatt_char

### DISCUSSION THREAD

**Discussion by ghost** [hbldh/bleak#1193] [Q&A] [answered]
Hi,
I am trying to use Bleak with Polar Verity Sense ([Link](https://github.com/polarofficial/polar-ble-sdk/blob/master/technical_documentation/Polar_Measurement_Data_Specification.pdf) to documentation) to stream ACC Data. I searched thoroughly in the issues and discussions threads and could not find any working sample. So I tried to write my own simple code. I am trying to produce the process highlighted below:

![Presentation1](https://user-images.githubusercontent.com/42944920/211008192-8f09b06d-3042-4e7e-aa2d-95bd93eda7bc.jpg)

The response that is recorded is not what I expect as seen below. What are the extra characters that I cannot decode? Am I doing something wrong? Is there any working example of a similar process?

Response 1: (b'\x02\xc3\x95\x9e\xc8"0\x14\n\x80\xdd\xfd\xbe\xff\xd3\xef\x06\x05\x02\xcfK\xb13C\x02\xf1\xe6\xc5\xf0\x02\x08\x14\xf6\x01\x10\x07\xff\xfd\xfd\xff\x08\xf8\xff\xff\x07!\x07\xe4\xec\x03%\xea\xee\xf8\x17\x02\x06\xfd\xf9\x07\x03\x03\xd9\x0c\x05\'\xe8\x00\xf5\xf7\x06\xef\x08\xfa\x06\xf1\xf4\x0b\x16\x07\xf7\xf5\x04\x0e\x07\xf6\x03\x10\x0b\xf8\xf2\x02\x06\x14\x00n\x0bJ\x00\xd8;\xb3\xf86l\xcd\x9aR\xb0h\x82\x1fy\x11(\x82a\xef6r\xe1=\x9d\xf0\x06\x02\xef\xbaE\x17\xc3\xc0\xfc\xc2\xa1\xc3+\x81(\x08\x14\x02\x1c\xf3\t\xf3\xfd\xf6\r\xf6\x05\xeb\x08\xe8\n\x18\xf3\xf1\xe4#\x15\x02\xf8\x03\xf8\x07\xe9\x12\x00\x06\xff\xf7\x15\x05\x04\xe9\x04\xfb\t\xf7\xff\xfe\x04\x08\xfb\xee\x05\x07\x13\xf9\x02\xf3\x05\xfc\x01\x07\xfb\xf8\xfe\xfb\x00\x06\x0b\xb8\xb3\xf8z\x01\x10\xb4\xde@\xc2\xdd\xff\x87\x8c\xd8\x05|\x1c\xcc_\xe7R_\xb8\x0b')

Response 2: (b'\x02\x18A8[\xa7/\x14\n\x80f\xff\xb7\xfd\xd5\xef\x08"\xfd\xf3\xf1\x11\xe3\x11\xf7\x05\xf6\x11\xfd\x08\xfd\x06\xf9\x04\xea\x08\xee\x01\xf5\x01\x16!\x1a\x05\xe8\x13\xf7\x06\x04\xeb\x02\n\x02\x14\x08\xe5\xec\xfb\x0c\n\x01\xfd\xff\x0b*\xec\x05\xf0\x05\xee\xfa\xfc\xef\xf6\x10\x03\n\xef)\xf0\x02\n\xda\x1c\'\x18\xfd\xfe\xe6\x03\xe9\x05\x03\xee\xfa\xed\x14\x08\x00\x10\r\x08\x05\xea\x05\xfe\xf0\xfe\x03\x15\xf3\xf1\n\r\xff\xf6\xff\x0c\x1d\xfa\x06\x14R\xba\xeb\xc1s\xd8\xfa\xde\xd4\xb6\xbd I\xe0K\x87-\xbb:)\x08Y\xef\xf7\xd4\x1d\xf7\x81\x12\xd2\x12\xe6\xbf\x80N@\x04=\x08\x01\xcbL\xbe\x9f\x0f\x08\x14\x05\xf3\xfd\xe7\x0b\xe8\xfe\xfe!\x06\t\x03\x01\xf7\xf2\x05\xf9\x06\x0c\x11\xfc\xea\x13\x03\x13\xf1\x00\x0b\xf3\xfb\x12\x11\xfa\xe3\xf4\x0c\xe9\x06\xfa\x0f\xfb\xed\x14\xfe\x15\x05\x1b\x00\xe4\xe0\x0e\x03\xff\xe3\x0e\x0e\x0b\xfb\x07\x0f')

**Comment by dlech** [comment, marked as answer]
It looks correct to me. You can make it look more like the docs by calling the `.hex(" ")` method on the bytes.

Or more usefully, use `struct.unpack_from()`.

```python
acc, timestamp, frame_type, ref0, ref1, ref2, size, count = struct.unpack_from("<BQB3hB", data)

base_offset = struct.calcsize("<BQB3hB")

for i in range(count):
    offset = base_offset + count * 3
    sample = data[offset:offset + 3]
    ...
```
